### PR TITLE
Deprecate OMP from Python package

### PR DESCRIPTION
**Pull request by RandySheriffH**
Per team's decision, deprecate OMP from packaging to prepare for coming release:
1. For previous openmp build, remove --use_openmp, so thread pool will become default;
2. For previous non-openmp build, add --use_openmp and rename the package to indicate the inclusion.
3. Add a mac build with openmp enabled.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 1]
```diff
@@ -167,8 +166,9 @@ stages:
                   --parallel \
                   --enable_lto \
                   --build_wheel \
+                  --use_openmp \
                   --enable_onnx_tests \
-                  --wheel_name_suffix=noopenmp \
+                  --wheel_name_suffix=openmp \
```
Comment (yuslepukhin) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:171]: >-wheel_name_suffix=openmp [](start = 19, length = 25)

So in some cases you add openmp.  I think the fact needs to be reflected in the PR description.

Comment (RandySheriffH) [reply tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:171]: Thanks! How about now?

**Comment by snnn** [comment]
Please also add a package for MacOS openmp.  And, in the noopenmp case, please remove the commands for installing openmp from brew.

**Comment by snnn** [comment]
As we talked offline:

In total we should have 5 different nuget variants:
1.	CPU with openmp
2.	CPU without openmp
3.	GPU
4.	CPU without contribops and without openmp
5.	Windows AI


And we have 8 packaging piplines. Below are my suggestions:
1. Node.js binding  : Need to remove openmp
2. Nuget GPU  (OnnxRuntime-Github): no need to change
3. Nuget WindowsAI: no need to change
4. Python packaging: update the package names and add MacOS
5. Python Packaging    (Training) : no need to change
6. Zip-Nuget-Java Packaging:  rename the pipeline name and packages to openmp.  And move the GPU part to NoOpenMP pipeline.
7. Zip-Nuget-Java Packaging    NoOpenMP: add GPU part from the noopenmp pipeline.
8. Zip-Nuget-Java Packaging    NoContribOps: no need to change.

Please correct me if I was wrong.

And I suggest you split this PR to 3: nodejs/python/nuget. So that we can validate them separately.

**Comment by snnn** [comment]
LGTM.

**Review by snnn — APPROVED** [review]

**Pull request merged**